service/src/main/java/bio/terra/cbas/dependencies/wes/CromwellService.java

```diff
@@ -128,6 +126,46 @@ public static String getErrorMessage(List<FailureMessage> failureMessages) {
     }
   }
 
+  /**
+   * Cromwell accepts an object "Workflow Options" to specify additional configuration for a
+   * workflow. Here, we build that object with the parameters we care about. final_workflow_log_dir
+   * specifies the path where outputs will be written. write_to_cache and read_from_cache are both
+   * related to call caching. Users expect that we will always write_to_cache, but only
+   * read_from_cache when call caching is enabled. This is how it works in GCP, so we are mirroring
+   * the behavior here. write_from_cache should always be true, and read_from_cache should only be
+   * true when the user enables call caching.
+   * https://cromwell.readthedocs.io/en/stable/wf_options/Overview/ for more info.
+   *
+   * @param isCallCachingEnabled Whether the user wishes to run this workflow with call caching.
+   * @return A string formatted as a JSON object that can be used as cromwell's Workflow Options.
+   * @throws JsonProcessingException
+   */
+  public String buildWorkflowOptionsJson(Boolean isCallCachingEnabled) {
+    // Map we will convert to JSON
+    Map<String, Object> workflowOptions = new HashMap<>();
+
+    // Path for cromwell to write workflow logs to.
+    Optional<String> finalWorkflowLogDir = cromwellClient.getFinalWorkflowLogDirOption();
+    if (finalWorkflowLogDir.isPresent()) {
+      workflowOptions.put("final_workflow_log_dir", finalWorkflowLogDir.get());
+    }
+
+    // Call caching fields
+    workflowOptions.put("write_to_cache", true);
+    workflowOptions.put("read_from_cache", isCallCachingEnabled);
```

Comment:
It seems like surely there is some way to use a `Map` literal as the basis for  the JSON, instead of side-effecting.